ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -57,11 +58,41 @@ message Commands {
   // Required
   repeated Command commands = 8;
 
-  // The length of the time window during which all commands with the same ``act_as`` parties and
-  // the same command ID will be deduplicated.
-  // Duplicate commands submitted before the end of this window return an ALREADY_EXISTS error.
+  // Specifies the deduplication period for the change ID.
+  // Deduplication provides the following deduplication guarantee:
+  // If the completion for this submission is a definite answer (as defined in ``completion.proto``), say at the completion offset ``off``,
+  // then there is no accepting completion for the same change ID between the start of the deduplication period and ``off``.
+  //
+  // The submission will be rejected if the deduplication period starts in the future
+  // (e.g., by specifying an offset that is higher than the completion end)
+  // or too early (e.g., by specifying an offset before the completion begin).
+  //
+  // If omitted, the participant chooses the start of the deduplication period at its discretion.
   // Optional
-  google.protobuf.Duration deduplication_time = 9;
+  oneof deduplication {
+    // Same semantics as deduplication_since
+    //
+    // DEPRECATED
+    google.protobuf.Duration deduplication_time = 9;
+
+    // Specifies the start of the deduplication period by a completion stream offset.
+    //
+    // Must be a valid LedgerString (as described in ``value.proto``).
+    string deduplication_offset = 14;
+
+    // Specifies a point in time at which the deduplication shall start.
+    // The participant or Daml ledger converts this timestamp into a completion offset using its local clock.
+    // The clock for this purpose is typically not fully synchronized with the application’s clock or record time,
+    // and message and processing delays may further increase the difference in interpretation.
+    google.protobuf.Timestamp deduplication_start = 15;
+
+    // Specifies the length of the deduplication period.
+    // The participant or Daml ledger converts the duration into a timestamp ``deduplication_start`` since
+    // by subtracting the duration from its local clock at some point during processing the submission.
+    //
+    // Must be non-negative.
+    google.protobuf.Duration deduplication_since = 16;
```

Comment:
```suggestion
    google.protobuf.Duration deduplication_duration = 16;
```
I noticed that it is easier to talk about duration and explain that they are applied backwards than talking about a "deduplication-since" value or the like.

Reply:
`deduplication_since` isn't a good name. In the Read/WriteService API, I've used the term `deduplication_window` for this, but `deduplication_duration` also sounds reasonable. Changed to `duration` for now. Please shout if you think `window` is better.

Reply:
Thanks. The write-up for KV ledgers worked well with duration.